### DISCUSSION TITLE
Scratchy Bluetooth audio output

### DISCUSSION THREAD

**Discussion by AydySmith** [pschatzmann/arduino-audio-tools#181] [Q&A] [answered]
I got my speaker working nicely using a Bluetooth input and writing it to an I2S Stream, there were no issues with the playback. So I decided to start adding filters (As directed in the documentation) and the audio has become scratchy (for lack of a better word). it almost sounds like there is a very small delay between bits being output and this makes the music almost bounce if that makes sense. I'm not too sure what I am doing wrong. 

I first define what I need for the filter(using the values from the docs as I have no frame of reference yet):
`FilteredStream<int16_t, float> filtered(i2s, channels);
StreamCopy copier(i2s, filtered);
float coef[] = {0.021, 0.096, 0.146, 0.096, 0.021};`

Then I call this filter in setup:
` filtered.setFilter(0, new FIR<float>(coef));
  filtered.setFilter(1, new FIR<float>(coef));`

**Comment by pschatzmann** [comment, marked as answer]
I haven't really used this functionality. I suggest that you have a look at [ESP32 Arduino with arduino audio tools - Using Filters](https://youtu.be/D0EUDRo4Mx0) from NA5Y and that you use https://fiiir.com/ to determine your parameters

**Reply by AydySmith** [reply]
Thank you, I took a look at the video, and it will be of help down the line. I have played around with the code to try to get things working and now the ESP32 is just continuously rebooting:

`rst:0x3 (SW_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1284
load:0x40078000,len:12808
load:0x40080400,len:3032
entry 0x400805e4
ets Jun  8 2016 00:22:57`

So I need to figure that out first, then I can get back to the filters. My best guess is that it's something to do with the A2DPStream class because there hasn't been an issue when using I2SStream with a2dp_sink, so I might have to move back to that.
